treesource.c

```diff
@@ -87,7 +87,7 @@ static void write_propval_string(FILE *f, const char *s, size_t len)
 			fprintf(f, "\\\"");
 			break;
 		case '\0':
-			fprintf(f, "\\0");
+			fprintf(f, "\", \"");
```

Comment:
As noted on the mailing list, this breaks a testcase which will need to be fixed to match.